File: frontend/src/constants/paymentModes.js
/** Align with backend customer_payment_mode_type / payment_mode_type */
export const CUSTOMER_PAYMENT_MODE_OPTIONS = [
  { value: "CASH", label: "Cash" },
  { value: "UPI", label: "UPI" },
  { value: "CARD", label: "Card" },
  { value: "CHEQUE", label: "Cheque" },
  { value: "NEFT", label: "NEFT" },
  { value: "OTHER", label: "Other" },
];

export function formatPaymentModeLabel(mode) {
  const m = String(mode || "").toUpperCase();
  if (!m || m === "CREDIT") return "Credit";
  const hit = CUSTOMER_PAYMENT_MODE_OPTIONS.find((o) => o.value === m);
  return hit?.label || m;
}

/** Default: counter / walk-in / cash customers pay at confirm unless user chooses credit. */
export function defaultCollectPaymentNow({ isRetailer, isWalkIn, isCashCustomer }) {
  if (isRetailer && isWalkIn) return true;
  if (isCashCustomer) return true;
  return false;
}

File: mobile/lib/core/constants/payment_modes.dart
/// Payment modes aligned with backend enums.
const List<String> kCustomerPaymentModes = [
  'CASH',
  'UPI',
  'CARD',
  'CHEQUE',
  'NEFT',
  'OTHER',
];

String formatPaymentModeLabel(String? mode) {
  final m = (mode ?? '').toUpperCase();
  if (m.isEmpty || m == 'CREDIT') return 'Credit';
  switch (m) {
    case 'CASH':
      return 'Cash';
    case 'UPI':
      return 'UPI';
    case 'CARD':
      return 'Card';
    case 'CHEQUE':
      return 'Cheque';
    case 'NEFT':
      return 'NEFT';
    case 'OTHER':
      return 'Other';
    default:
      return m;
  }
}

/// Default pay-now for walk-in / counter sales on mobile when wired into billing UI.
bool defaultCollectPaymentNow({
  required bool isRetailer,
  required bool isWalkIn,
  required bool isCashCustomer,
}) {
  if (isRetailer && isWalkIn) return true;
  if (isCashCustomer) return true;
  return false;
}
